File: pkg/dart2cpp/sample/dart/test_function_param.dart
/// 测试函数作为参数
int add(int a, int b) {
  return a + b;
}

int subtract(int a, int b) {
  return a - b;
}

int calculate(int a, int b, int Function(int, int) operation) {
  return operation(a, b);
}

void main() {
  int result1 = calculate(10, 5, add);
  print('10 + 5 = $result1');

  int result2 = calculate(10, 5, subtract);
  print('10 - 5 = $result2');
}

File: pkg/dart2cpp/sample/output/test_function_param.cpp
#include "dart2cpp.h"

// 工具宏定义

Int add(Int a, Int b);
Int subtract(Int a, Int b);
Int calculate(Int a, Int b, ObjectPtr<Function> operation);
Int add(Int a, Int b) {
  return (a + b);
}

Int subtract(Int a, Int b) {
  return (a - b);
}

Int calculate(Int a, Int b, ObjectPtr<Function> operation) {
  return Int(operation->apply({a, b}));
}

// ============================================================================
// 主函数
// ============================================================================

int main() {
  try {
    auto result1 = calculate(dart_int(10), dart_int(5), makeFunction(&add));
dart_print(dart_concat(dart_string("10 + 5 = "), result1));
auto result2 = calculate(dart_int(10), dart_int(5), makeFunction(&subtract));
dart_print(dart_concat(dart_string("10 - 5 = "), result2));
    return 0;
  } catch (const std::exception& e) {
    std::cerr << "Error: " << e.what() << std::endl;
    return 1;
  }
}
